custom/icds_reports/filters.py

```diff
@@ -37,38 +37,63 @@ def loc_to_json(loc, project):
 
     project = Domain.get_by_name(domain)
 
-    locations = SQLLocation.root_locations(domain)
+    if user.has_permission(domain, 'access_all_locations'):
+        accessible_root_locations = SQLLocation.root_locations(domain)
+    else:
+        accessible_root_locations = [(user.sql_location
+                                      .get_ancestors(include_self=True)
+                                      .get(location_type__code='state'))
+                                     ]
+
     if not show_test:
-        locations = [
-            loc for loc in locations if loc.metadata.get('is_test_location', 'real') != 'test'
-        ]
+        accessible_root_locations = [loc for loc in accessible_root_locations
+                                     if loc.metadata.get('is_test_location', 'real') != 'test']
 
-    loc_json = [loc_to_json(loc, project) for loc in locations]
+    loc_json = [loc_to_json(loc, project) for loc in accessible_root_locations]
 
     # if a location is selected, we need to pre-populate its location hierarchy
     # so that the data is available client-side to pre-populate the drop-downs
     if selected_loc_id:
+        if not user.has_permission(domain, 'access_all_locations'):
+            primary_location = user.sql_location
```

Comment:
`user.sql_location` is not a cached property, not sure why though. So this would fire up a query each time, Fetching this on top once would be nice.

Reply:
make sense